.github/workflows/create-release-tag.yml

```diff
@@ -0,0 +1,110 @@
+name: Create Tag and Release with Changelog
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag_name:
+        description: 'Tag name for the new release'
+        required: true
+
+permissions:
+  contents: write
+  packages: write
+  pull-requests: write
+  
+jobs:
+  fetch-tag:
+    runs-on: ubuntu-latest
+    outputs:
+      old_tag: ${{ steps.get_tag.outputs.old_tag_name }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
```

Comment:
Would it be better to fetch the tags using the GitHub API, so that there is no need to checkout the code?

Reply:
For time being is this okay ? or do you want me to change this ?

Reply:
No need to do in this PR, but it would be nice as a follow-up enhancement.